cleanlab/filter.py

```diff
@@ -452,12 +452,13 @@ class 0, 1, ..., K-1.
     if filter_by == "predicted_neq_given":
         label_issues_mask = find_predicted_neq_given(labels, pred_probs, multi_label=multi_label)
 
-    # Remove label issues if given label == model prediction
-    # TODO: consider use of _multiclass_crossval_predict() here
-    pred = pred_probs.argmax(axis=1)
-    for i, pred_label in enumerate(pred):
-        if pred_label == labels[i]:
-            label_issues_mask[i] = False
+    if filter_by not in ["low_self_confidence", "low_normalized_margin"]:
+        # Remove label issues if given label == model prediction if issues haven't been removed yet
+        # TODO: consider use of _multiclass_crossval_predict() here
```

Comment:
```suggestion
```

Reply:
we no longer are considering have filter methods generate predictions, that's what distinguishes them from cleanlearning methods and others that can generate predictions